---
title: "recount3: uniformly processed RNA-seq"
image: "logos/recount3-square.png"
links:
  - label: Documentation
    url: "docs/"
  - label: Study Explorer
    url: "https://jhubiostatistics.shinyapps.io/recount3-study-explorer/"
output:
  postcards::trestles
---

<img src="logos/logo.png" width="500px"/>

**recount3** is an online resource consisting of RNA-seq gene, exon, and exon-exon junction counts as well as coverage bigWig files for 8,679 and 10,088 different studies for human and mouse respectively. It is the third generation of the [ReCount project](http://bowtie-bio.sourceforge.net/recount/) and part of [**recount.bio**](https://recount.bio).

The raw sequencing data were processed with the Monorail system as described in the [recount3 paper](TODO) which created the coverage bigWig files and the recount-unified text files. While these raw output files are available through [IDIES SciServer](https://idies.jhu.edu/), for ease of statistical analysis, we provide through the `r BiocStyle::Biocpkg("recount3")` R/Bioconductor package an interface that builds [RangedSummarizedExperiment](https://doi.org/10.1038/nmeth.3252) R objects for gene, exon, and exon-exon junction counts for each study. Furthermore, `r BiocStyle::Biocpkg("snapcount")` enables query-based access of the `recount3` and `recount2` data. The coverage bigWig files can be used for annotation-agnostic expression analyses using for example `r BiocStyle::Biocpkg("megadepth")`, `r BiocStyle::Biocpkg("derfinder")` and other tools. 

By taking care of several pre-processing steps and combining many datasets into one easily-accessible website, we make finding and analyzing RNA-seq data considerably more straightforward. For more details about `recount3`, check the [documentation book](docs/).

## Study explorer

You can open the study explorer independently through [shinyapps.io](https://jhubiostatistics.shinyapps.io/recount3-study-explorer/) to explore the data hosted by the `recount3` project.

## Related publications

Additional publications directly related to `recount3` are:

* Megadepth: efficient coverage quantification for BigWigs and BAMs. Christopher Wilks, Omar Ahmed, Daniel N. Baker, David Zhang, Leonardo Collado-Torres, Ben Langmead. bioRxiv, 2020. DOI: [10.1101/2020.12.17.423317](https://doi.org/10.1101/2020.12.17.423317)

## Related projects

The `ReCount` project has three main publications over the years with implementation details that have been improved each time. Each phase of the project has had several companion publications that expanded the main resource in different directions. Please cite the appropriate expansions and results as you see relevant for your work.

Previous `ReCount` projects:

* [`recount2`](https://jhubiostatistics.shinyapps.io/recount/), DOI: [10.1038/nbt.3838](https://doi.org/10.1038/nbt.3838)
* [`ReCount`](http://bowtie-bio.sourceforge.net/recount/), DOI: [10.1186/1471-2105-12-449](https://doi.org/10.1186/1471-2105-12-449)

## Team members

* [Christopher Wilks](https://github.com/ChristopherWilks) (the star behind `recount3`!)
* [Leonardo Collado Torres](https://github.com/lcolladotor) 
* [Shijie Charles Zheng](https://github.com/sjczheng)
* [Kevin FY Chen](https://github.com/KevinFYChen)
* [Kasper Daniel Hansen](https://github.com/kasperdanielhansen)
* [Ben Langmead](https://github.com/BenLangmead)


## Teams involved

* [Ben Langmead's lab at JHU Computer Science](http://www.langmead-lab.org/)
* [Kasper Daniel Hansen's lab at JHBSPH Biostatistics Department](https://www.hansenlab.org/)
* [Leonardo Collado-Torres](http://lcolladotor.github.io/) and [Andrew E. Jaffe](http://aejaffe.com/) from [LIBD](https://www.libd.org/)
* [Abhinav Nellore's lab at OHSU](http://nellore.bio/)
* [Jeff Leek's lab at JHBSPH Biostatistics Deparment](http://jtleek.com/)
* Data hosted by [SciServer from IDIES at JHU](https://www.sciserver.org/)

