# Bioconductor

R is an open source software programming language that has been widely used for studying gene expression data thanks in part to the [Bioconductor project](http://bioconductor.org/). You can install R for free from [CRAN](https://cran.r-project.org/) on macOS, Windows, and Linux operating systems.

The Bioconductor community has defined common structures that enable users to store their data in common formats and analyzed through many different solutions (DOI: [10.1038/nmeth.3252](https://doi.org/10.1038/nmeth.3252)). One such common format is the `RangedSummarizedExperiment` object from the `r BiocStyle::Biocpkg("SummarizedExperiment")` package that can then be used in downstream analayses of differential expression using packages such as `r BiocStyle::Biocpkg("DESeq2")`, `r BiocStyle::Biocpkg("edgeR")`, and `r BiocStyle::Biocpkg("limma")`.

To facilitate the re-analysis of public expression data, we have provided several R/Bioconductor packages that interface with the `recount3` data. These are:

* `r BiocStyle::Biocpkg("recount3")`
* `r BiocStyle::Biocpkg("snapcount")`
* `r BiocStyle::Biocpkg("megadepth")`

These packages can also benefit from using `r BiocStyle::Biocpkg("recount")` and `r BiocStyle::Biocpkg("derfinder")`, which we made in earlier phases of the `ReCount` project.

In the Bioconductor landing pages for our packages you can find detailed vignette documents, such as the one called [`recount3` quick start guide](http://bioconductor.org/packages/release/bioc/vignettes/recount3/inst/doc/recount3-quickstart.html). Vignette documents are user guides illustrating how the different functions in a given R package can be used together. The code shown in this chapter is a subset from these vignette documents, thus we highly recommend that you read the vignettes if you want to learn more details about these R packages than what we have illustrated here. 

## recount3

_The following information is taken from the [`recount3`](https://bioconductor.org/packages/recount3) documentation._

After installing `r Biocpkg('recount3')`, we need to load the package, which will automatically load the required dependencies.

```{r 'start', message=FALSE}
## Load recount3 R package
library("recount3")
```

If you have identified a **study** of interest and want to access the gene level expression data, use `create_rse()` as shown below. `create_rse()` has arguments that will allow you to specify the  **annotation** of interest for the given organism, and whether you want to download **gene**, **exon** or **exon-exon junction** expression data. 


```{r 'quick_example'}
## Find all available human projects
human_projects <- available_projects()

## Find the project you are interested in,
## here we use SRP009615 as an example
proj_info <- subset(
    human_projects,
    project == "SRP009615" & project_type == "data_sources"
)

## Create a RangedSummarizedExperiment (RSE) object at the gene level
rse_gene_SRP009615 <- create_rse(proj_info)

## Explore that RSE object
rse_gene_SRP009615
```

You can also interactively choose your study of interest

```{r "interactive_display", eval = FALSE}
## Note that you can interactively explore the available projects
proj_info_interactive <- interactiveDisplayBase::display(human_projects)

## Select a single row, then hit "send". The following code checks this.
stopifnot(nrow(proj_info_interactive) == 1)

## Then create the RSE object
rse_gene_interactive <- create_rse(proj_info_interactive)
```

Once you have a RSE file, you can use `transform_counts()` to transform the raw coverage counts.


```{r "tranform_counts"}
## Once you have your RSE object, you can transform the raw coverage
## base-pair coverage counts using transform_counts().
## For RPKM, TPM or read outputs, check the details in transform_counts().
assay(rse_gene_SRP009615, "counts") <- transform_counts(rse_gene_SRP009615)
```

Now you are ready to continue with downstream analysis software.


`r Biocpkg('recount3')` also supports accessing the BigWig raw coverage files as well as specific study or collection sample **metadata**. Please continue to the [users guide](https://bioconductor.org/packages/recount3) for more detailed information.

## snapcount

_The following information is taken from the [`snapcount`](https://bioconductor.org/packages/snapcount) documentation. As stated in the documentation, `snapcount` provides access to data from the `recount2` project, but it can also be used with the data from the `recount3` project._

To get started, we need to load the `r Biocpkg("snapcount")` package into our R session. This will load all the required dependencies.

```{r "start_snapcount", message=FALSE}
library("snapcount")
```

snapcount makes it easy to query the [Snaptron web services](http://snaptron.cs.jhu.edu/), with results presented as `r Biocpkg('RangedSummarizedExperiment')` objects.
You can query measurements for **genes**, **exons**, **splice junctions** and **coverage vectors** from the RNA-seq samples indexed in Snaptron.
Samples are organized into compilations (e.g. `srav2`) that altogether contain the same studies and summaries in the `r Biocpkg('recount')` resource.  Queries can be filtered to narrow the focus to particular genes or genomic intervals, to events with certain prevalence, to events that do or don't appear in gene annotation, or to samples with particular metadata.

snapcount complements the `r Biocpkg('recount')` package, which also allows for searching by coordinates or by HUGO gene names.
In general, `r Biocpkg('recount')` works best when you are interested in *all* the genes, exons, or splice junctions in a study, whereas snapcount is best for queries over a particular subset of genes or intervals across all or most of the samples in recount2/Snaptron.  The more specific your query, the faster and easier it will be to use snapcount.

All the RNA-seq samples in the recount and Snaptron resources were analyzed using the [Rail-RNA](http://rail.bio) aligner.  Rail produces spliced alignments that in turn produce coverage and junction-level summaries that are further processed to obtain the data in recount and Snaptron.

Similar to the `r Biocpkg('recount')` all coverage counts in Snaptron/snapcount are stored/retrieved as raw, un-normalized counts.

### Basic queries

A basic query returns a *RangedSummarizedExperiment* (RSE) object with one or more features (genes, exons, or splice junctions) as *rowRanges*.
Raw coverage counts are returned as the *counts* assay in the RSE  Full sample metadata is returned as the  *colData* of the RSE

Basic queries include:

* exon-exon splice junction counts: *query_jx*
* exon-level quantifications: *query_exon*
* genes-level quantifications: *query_gene*

The Gencode v25 annotation defines what genes and exons can be queried (as in `r Biocpkg('recount')`).
For splice junctions, both annotated and novel junctions are queried at the same time unless one is explicitly filtered.

Metadata columns will vary by compilation (e.g. TCGA vs. GTEx).
Please be cautious with metadata.  We strive to make it as complete and usable as possible, but it can still be incomplete, incorrect or inconsistently formatted (e.g. "age" in the `srav2` compilation).

Start by querying for all junctions within the region of gene CD99:
```{r 'snapcount_gene', eval = TRUE}
##Query coverage for gene, exon, and annotated junctions across all
#in the region of the CD99 gene
#from GTEx v6 sample compilation
#CD99 is chosen for its size
sb <- QueryBuilder(compilation="gtex", regions="CD99")
cd99.gene <- query_gene(sb)
dim(cd99.gene)
head(cd99.gene)
```

Now query for junctions:
```{r 'snapcount_jx', eval = TRUE}
##Query all exon-exon splice junctions within the region of gene CD99
cd99.jx.all <- query_jx(sb)
dim(cd99.jx.all)
cd99.jx.all
```

Now query for junctions and filter by sample type: Brain:
```{r 'snapcount_jx2', eval = TRUE}
#now subfilter by sample tissue
#GTEx samples that are labeled with tissue type "Brain"
sb <- set_column_filters(sb, SMTS == "Brain")
cd99.jx.all <- query_jx(sb)
dim(cd99.jx.all)
head(cd99.jx.all)
```

Same query/filter again but for exons:
```{r 'snapcount_exon', eval = TRUE}
cd99.exon <- query_exon(sb)
dim(cd99.exon)
head(cd99.exon)
```

Now query junctions but further filter for only those which are fully annotated:
```{r 'snapcount_jx3', eval = TRUE}
###Only query junctions which are fully annotated---both left and
#right splice sites are found together in one or more of the
#Snaptron sourced annotations
sb <- set_row_filters(sb, annotated == 1)
cd99.jx <- query_jx(sb)
dim(cd99.jx)
head(cd99.jx)
```

Now check an example of the metadata stored in the RSE, InsertSize:
```{r 'snapcount_md', eval = TRUE}
##Metadata is stored directly in the RSE object.
#For example the library insert size can be retrieved
#across all runs  in the RSE
head(cd99.jx.all$InsertSize)
```

## megadepth

_The following information is taken from the [`megadepth`](https://bioconductor.org/packages/megadepth) documentation._

To get started, we need to load the `r Biocpkg("megadepth")` package into our R session. This will load all the required dependencies.

```{r "start_megadepth", message=FALSE}
library("megadepth")
```

Once we have the R package loaded, we need to install the [Megadepth](https://github.com/ChristopherWilks/megadepth) software. We can do so with `install_megadepth()`, which downloads a binary for your OS (Linux, Windows or macOS) ^[Please check [Megadepth](https://github.com/ChristopherWilks/megadepth) for instructions on how to compile the software from source if the binary version doesn't work for you.]. We can then use with an example BigWig file to compute the coverage at a set of regions.

```{r "install_software"}
## Install the latest version of Megadepth
install_megadepth(force = TRUE)
```

Next, we might want to use `r Biocpkg("megadepth")` to quantify the coverage at a set of regions of the genome of interest to us. Here we will use two example files that are include in the package for illustration and testing purposes. One of them is a [bigWig file](https://genome.ucsc.edu/goldenPath/help/bigWig.html) that contains the base-pair coverage information for a sample of interest and the second one is [BED file](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) which contains the genomic region coordinates of interest. So we first locate them.

```{r "locate_example_bw"}
## Next, we locate the example BigWig and annotation files
example_bw <- system.file("tests", "test.bam.all.bw",
    package = "megadepth", mustWork = TRUE
)
annotation_file <- system.file("tests", "testbw2.bed",
    package = "megadepth", mustWork = TRUE
)

## Where are they?
example_bw
annotation_file
```

Once we have located the example files we can proceed to calculating the base-pair coverage for our genomic regions of interest. There are several ways to do this with `r Biocpkg("megadepth")`, but here we use the user-friendly function `get_coverage()`. This function will perform a given operation **op** on the bigWig file for a given set of regions of interest (_annotation_). One of those operations is to compute the mean base-pair coverage for each input region. This is what we'll do with our example bigWig file.

```{r example}
## We can then use megadepth to compute the coverage
bw_cov <- get_coverage(
    example_bw,
    op = "mean",
    annotation = annotation_file
)
bw_cov
```

`get_coverage()` returns an object that is familiar to `r Biocpkg("GenomicRanges")` users, that is, a `GRanges` object that can be used with other Bioconductor software packages.

This example is just the tip of the iceberg, as [Megadepth](https://github.com/ChristopherWilks/megadepth) and thus `r Biocpkg("megadepth")` can do a lot of useful processing operations on BAM and bigWig files.

Please continue to the [users guide](https://bioconductor.org/packages/megadepth) for more detailed information.

