<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Bioconductor | recount3: human and mouse public RNA-seq data</title>
  <meta name="description" content="5 Bioconductor | recount3: human and mouse public RNA-seq data" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Bioconductor | recount3: human and mouse public RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Bioconductor | recount3: human and mouse public RNA-seq data" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="quality-check-fields.html"/>
<link rel="next" href="raw-files.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="css/styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="part"><span><b>I Get RNA-seq data!</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#human"><i class="fa fa-check"></i><b>1.1</b> Human</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#mouse"><i class="fa fa-check"></i><b>1.2</b> Mouse</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#study-explorer"><i class="fa fa-check"></i><b>1.3</b> Study explorer</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#how-to-cite-recount3"><i class="fa fa-check"></i><b>1.4</b> How to cite <code>recount3</code></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quick-access.html"><a href="quick-access.html"><i class="fa fa-check"></i><b>2</b> Quick Access</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-access.html"><a href="quick-access.html#quick-recount3"><i class="fa fa-check"></i><b>2.1</b> Quick recount3</a></li>
<li class="chapter" data-level="2.2" data-path="quick-access.html"><a href="quick-access.html#quick-snapcount"><i class="fa fa-check"></i><b>2.2</b> Quick snapcount</a></li>
<li class="chapter" data-level="2.3" data-path="quick-access.html"><a href="quick-access.html#quick-raw-files"><i class="fa fa-check"></i><b>2.3</b> Quick raw files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html"><i class="fa fa-check"></i><b>3</b> How to ask for help</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#gitter"><i class="fa fa-check"></i><b>3.1</b> Gitter</a></li>
<li class="chapter" data-level="3.2" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#bioconductor-support-website"><i class="fa fa-check"></i><b>3.2</b> Bioconductor Support website</a></li>
<li class="chapter" data-level="3.3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#github-issues"><i class="fa fa-check"></i><b>3.3</b> GitHub Issues</a></li>
</ul></li>
<li class="part"><span><b>II Details</b></span></li>
<li class="chapter" data-level="4" data-path="quality-check-fields.html"><a href="quality-check-fields.html"><i class="fa fa-check"></i><b>4</b> Quality check fields</a></li>
<li class="chapter" data-level="5" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>5</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="5.1" data-path="bioconductor.html"><a href="bioconductor.html#recount3"><i class="fa fa-check"></i><b>5.1</b> recount3</a></li>
<li class="chapter" data-level="5.2" data-path="bioconductor.html"><a href="bioconductor.html#snapcount"><i class="fa fa-check"></i><b>5.2</b> snapcount</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="bioconductor.html"><a href="bioconductor.html#basic-queries"><i class="fa fa-check"></i><b>5.2.1</b> Basic queries</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="bioconductor.html"><a href="bioconductor.html#megadepth"><i class="fa fa-check"></i><b>5.3</b> megadepth</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="raw-files.html"><a href="raw-files.html"><i class="fa fa-check"></i><b>6</b> Raw Files</a>
<ul>
<li class="chapter" data-level="6.1" data-path="raw-files.html"><a href="raw-files.html#data-source-vs-collection"><i class="fa fa-check"></i><b>6.1</b> Data source vs collection</a></li>
<li class="chapter" data-level="6.2" data-path="raw-files.html"><a href="raw-files.html#annotation-files"><i class="fa fa-check"></i><b>6.2</b> Annotation files</a></li>
<li class="chapter" data-level="6.3" data-path="raw-files.html"><a href="raw-files.html#project-level-count-files"><i class="fa fa-check"></i><b>6.3</b> Project-level count files</a></li>
<li class="chapter" data-level="6.4" data-path="raw-files.html"><a href="raw-files.html#project-level-metadata-files"><i class="fa fa-check"></i><b>6.4</b> Project-level metadata files</a></li>
<li class="chapter" data-level="6.5" data-path="raw-files.html"><a href="raw-files.html#sample-level-bigwig-files"><i class="fa fa-check"></i><b>6.5</b> Sample-level BigWig files</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="related-software.html"><a href="related-software.html"><i class="fa fa-check"></i><b>7</b> Related software</a>
<ul>
<li class="chapter" data-level="7.1" data-path="related-software.html"><a href="related-software.html#beyond-recount3"><i class="fa fa-check"></i><b>7.1</b> Beyond recount3</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="related-analyses.html"><a href="related-analyses.html"><i class="fa fa-check"></i><b>8</b> Related analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="related-analyses.html"><a href="related-analyses.html#beyond-recount3-1"><i class="fa fa-check"></i><b>8.1</b> Beyond recount3</a></li>
</ul></li>
<li class="part"><span><b>III Expand <code>ReCount</code></b></span></li>
<li class="chapter" data-level="9" data-path="your-data.html"><a href="your-data.html"><i class="fa fa-check"></i><b>9</b> Your data</a>
<ul>
<li class="chapter" data-level="9.1" data-path="your-data.html"><a href="your-data.html#process-your-data"><i class="fa fa-check"></i><b>9.1</b> Process your data</a></li>
<li class="chapter" data-level="9.2" data-path="your-data.html"><a href="your-data.html#contribute-your-collections"><i class="fa fa-check"></i><b>9.2</b> Contribute your collections</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="your-data.html"><a href="your-data.html#collection-metadata"><i class="fa fa-check"></i><b>9.2.1</b> Collection metadata</a></li>
<li class="chapter" data-level="9.2.2" data-path="your-data.html"><a href="your-data.html#collection-project-location"><i class="fa fa-check"></i><b>9.2.2</b> Collection project location</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="join-the-recount-team.html"><a href="join-the-recount-team.html"><i class="fa fa-check"></i><b>10</b> Join the <code>ReCount</code> team!</a></li>
<li class="chapter" data-level="" data-path="r-session-information.html"><a href="r-session-information.html"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">recount3: human and mouse public RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioconductor" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Bioconductor</h1>
<p>R is an open source software programming language that has been widely used for studying gene expression data thanks in part to the <a href="http://bioconductor.org/">Bioconductor project</a>. You can install R for free from <a href="https://cran.r-project.org/">CRAN</a> on macOS, Windows, and Linux operating systems.</p>
<p>The Bioconductor community has defined common structures that enable users to store their data in common formats and analyzed through many different solutions (DOI: <a href="https://doi.org/10.1038/nmeth.3252">10.1038/nmeth.3252</a>). One such common format is the <code>RangedSummarizedExperiment</code> object from the <em><a href="https://bioconductor.org/packages/3.14/SummarizedExperiment">SummarizedExperiment</a></em> package that can then be used in downstream analayses of differential expression using packages such as <em><a href="https://bioconductor.org/packages/3.14/DESeq2">DESeq2</a></em>, <em><a href="https://bioconductor.org/packages/3.14/edgeR">edgeR</a></em>, and <em><a href="https://bioconductor.org/packages/3.14/limma">limma</a></em>.</p>
<p>To facilitate the re-analysis of public expression data, we have provided several R/Bioconductor packages that interface with the <code>recount3</code> data. These are:</p>
<ul>
<li><em><a href="https://bioconductor.org/packages/3.14/recount3">recount3</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.14/snapcount">snapcount</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.14/megadepth">megadepth</a></em></li>
</ul>
<p>These packages can also benefit from using <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em> and <em><a href="https://bioconductor.org/packages/3.14/derfinder">derfinder</a></em>, which we made in earlier phases of the <code>ReCount</code> project.</p>
<p>In the Bioconductor landing pages for our packages you can find detailed vignette documents, such as the one called <a href="http://bioconductor.org/packages/release/bioc/vignettes/recount3/inst/doc/recount3-quickstart.html"><code>recount3</code> quick start guide</a>. Vignette documents are user guides illustrating how the different functions in a given R package can be used together. The code shown in this chapter is a subset from these vignette documents, thus we highly recommend that you read the vignettes if you want to learn more details about these R packages than what we have illustrated here.</p>
<div id="recount3" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> recount3</h2>
<p><em>The following information is taken from the <a href="https://bioconductor.org/packages/recount3"><code>recount3</code></a> documentation.</em></p>
<p>After installing <em><a href="https://bioconductor.org/packages/3.14/recount3">recount3</a></em>, we need to load the package, which will automatically load the required dependencies.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="bioconductor.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load recount3 R package</span></span>
<span id="cb6-2"><a href="bioconductor.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;recount3&quot;</span>)</span></code></pre></div>
<p>If you have identified a <strong>study</strong> of interest and want to access the gene level expression data, use <code>create_rse()</code> as shown below. <code>create_rse()</code> has arguments that will allow you to specify the <strong>annotation</strong> of interest for the given organism, and whether you want to download <strong>gene</strong>, <strong>exon</strong> or <strong>exon-exon junction</strong> expression data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="bioconductor.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Find all available human projects</span></span>
<span id="cb7-2"><a href="bioconductor.html#cb7-2" aria-hidden="true" tabindex="-1"></a>human_projects <span class="ot">&lt;-</span> <span class="fu">available_projects</span>()</span></code></pre></div>
<pre><code>## 2021-09-15 14:16:30 caching file sra.recount_project.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/sra.recount_project.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:34 caching file gtex.recount_project.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/gtex/metadata/gtex.recount_project.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:37 caching file tcga.recount_project.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/tcga/metadata/tcga.recount_project.MD.gz&#39;</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="bioconductor.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Find the project you are interested in,</span></span>
<span id="cb14-2"><a href="bioconductor.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="do">## here we use SRP009615 as an example</span></span>
<span id="cb14-3"><a href="bioconductor.html#cb14-3" aria-hidden="true" tabindex="-1"></a>proj_info <span class="ot">&lt;-</span> <span class="fu">subset</span>(</span>
<span id="cb14-4"><a href="bioconductor.html#cb14-4" aria-hidden="true" tabindex="-1"></a>    human_projects,</span>
<span id="cb14-5"><a href="bioconductor.html#cb14-5" aria-hidden="true" tabindex="-1"></a>    project <span class="sc">==</span> <span class="st">&quot;SRP009615&quot;</span> <span class="sc">&amp;</span> project_type <span class="sc">==</span> <span class="st">&quot;data_sources&quot;</span></span>
<span id="cb14-6"><a href="bioconductor.html#cb14-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-7"><a href="bioconductor.html#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="bioconductor.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a RangedSummarizedExperiment (RSE) object at the gene level</span></span>
<span id="cb14-9"><a href="bioconductor.html#cb14-9" aria-hidden="true" tabindex="-1"></a>rse_gene_SRP009615 <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(proj_info)</span></code></pre></div>
<pre><code>## 2021-09-15 14:16:44 downloading and reading the metadata.</code></pre>
<pre><code>## 2021-09-15 14:16:44 caching file sra.sra.SRP009615.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/15/SRP009615/sra.sra.SRP009615.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:45 caching file sra.recount_project.SRP009615.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/15/SRP009615/sra.recount_project.SRP009615.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:46 caching file sra.recount_qc.SRP009615.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/15/SRP009615/sra.recount_qc.SRP009615.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:46 caching file sra.recount_seq_qc.SRP009615.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/15/SRP009615/sra.recount_seq_qc.SRP009615.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:47 caching file sra.recount_pred.SRP009615.MD.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/15/SRP009615/sra.recount_pred.SRP009615.MD.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:48 downloading and reading the feature information.</code></pre>
<pre><code>## 2021-09-15 14:16:48 caching file human.gene_sums.G026.gtf.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:49 downloading and reading the counts: 12 samples across 63856 features.</code></pre>
<pre><code>## 2021-09-15 14:16:49 caching file sra.gene_sums.SRP009615.G026.gz.</code></pre>
<pre><code>## adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/gene_sums/15/SRP009615/sra.gene_sums.SRP009615.G026.gz&#39;</code></pre>
<pre><code>## 2021-09-15 14:16:50 construcing the RangedSummarizedExperiment (rse) object.</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="bioconductor.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Explore that RSE object</span></span>
<span id="cb33-2"><a href="bioconductor.html#cb33-2" aria-hidden="true" tabindex="-1"></a>rse_gene_SRP009615</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 63856 12 
## metadata(8): time_created recount3_version ... annotation recount3_url
## assays(1): raw_counts
## rownames(63856): ENSG00000278704.1 ENSG00000277400.1 ...
##   ENSG00000182484.15_PAR_Y ENSG00000227159.8_PAR_Y
## rowData names(10): source type ... havana_gene tag
## colnames(12): SRR387777 SRR387778 ... SRR389077 SRR389078
## colData names(175): rail_id external_id ...
##   recount_pred.curated.cell_line BigWigURL</code></pre>
<p>You can also interactively choose your study of interest</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="bioconductor.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Note that you can interactively explore the available projects</span></span>
<span id="cb35-2"><a href="bioconductor.html#cb35-2" aria-hidden="true" tabindex="-1"></a>proj_info_interactive <span class="ot">&lt;-</span> interactiveDisplayBase<span class="sc">::</span><span class="fu">display</span>(human_projects)</span>
<span id="cb35-3"><a href="bioconductor.html#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="bioconductor.html#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Select a single row, then hit &quot;send&quot;. The following code checks this.</span></span>
<span id="cb35-5"><a href="bioconductor.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">nrow</span>(proj_info_interactive) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb35-6"><a href="bioconductor.html#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="bioconductor.html#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Then create the RSE object</span></span>
<span id="cb35-8"><a href="bioconductor.html#cb35-8" aria-hidden="true" tabindex="-1"></a>rse_gene_interactive <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(proj_info_interactive)</span></code></pre></div>
<p>Once you have a RSE file, you can use <code>transform_counts()</code> to transform the raw coverage counts.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="bioconductor.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Once you have your RSE object, you can transform the raw coverage</span></span>
<span id="cb36-2"><a href="bioconductor.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="do">## base-pair coverage counts using transform_counts().</span></span>
<span id="cb36-3"><a href="bioconductor.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="do">## For RPKM, TPM or read outputs, check the details in transform_counts().</span></span>
<span id="cb36-4"><a href="bioconductor.html#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(rse_gene_SRP009615, <span class="st">&quot;counts&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">transform_counts</span>(rse_gene_SRP009615)</span></code></pre></div>
<p>Now you are ready to continue with downstream analysis software.</p>
<p><em><a href="https://bioconductor.org/packages/3.14/recount3">recount3</a></em> also supports accessing the BigWig raw coverage files as well as specific study or collection sample <strong>metadata</strong>. Please continue to the <a href="https://bioconductor.org/packages/recount3">users guide</a> for more detailed information.</p>
</div>
<div id="snapcount" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> snapcount</h2>
<p><em>The following information is taken from the <a href="https://bioconductor.org/packages/snapcount"><code>snapcount</code></a> documentation. As stated in the documentation, <code>snapcount</code> provides access to data from the <code>recount2</code> project, but it can also be used with the data from the <code>recount3</code> project.</em></p>
<p>To get started, we need to load the <em><a href="https://bioconductor.org/packages/3.14/snapcount">snapcount</a></em> package into our R session. This will load all the required dependencies.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="bioconductor.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;snapcount&quot;</span>)</span></code></pre></div>
<p>snapcount makes it easy to query the <a href="http://snaptron.cs.jhu.edu/">Snaptron web services</a>, with results presented as <em><a href="https://bioconductor.org/packages/3.14/RangedSummarizedExperiment">RangedSummarizedExperiment</a></em> objects.
You can query measurements for <strong>genes</strong>, <strong>exons</strong>, <strong>splice junctions</strong> and <strong>coverage vectors</strong> from the RNA-seq samples indexed in Snaptron.
Samples are organized into compilations (e.g. <code>srav2</code>) that altogether contain the same studies and summaries in the <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em> resource. Queries can be filtered to narrow the focus to particular genes or genomic intervals, to events with certain prevalence, to events that do or don’t appear in gene annotation, or to samples with particular metadata.</p>
<p>snapcount complements the <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em> package, which also allows for searching by coordinates or by HUGO gene names.
In general, <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em> works best when you are interested in <em>all</em> the genes, exons, or splice junctions in a study, whereas snapcount is best for queries over a particular subset of genes or intervals across all or most of the samples in recount2/Snaptron. The more specific your query, the faster and easier it will be to use snapcount.</p>
<p>All the RNA-seq samples in the recount and Snaptron resources were analyzed using the <a href="http://rail.bio">Rail-RNA</a> aligner. Rail produces spliced alignments that in turn produce coverage and junction-level summaries that are further processed to obtain the data in recount and Snaptron.</p>
<p>Similar to the <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em> all coverage counts in Snaptron/snapcount are stored/retrieved as raw, un-normalized counts.</p>
<div id="basic-queries" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Basic queries</h3>
<p>A basic query returns a <em>RangedSummarizedExperiment</em> (RSE) object with one or more features (genes, exons, or splice junctions) as <em>rowRanges</em>.
Raw coverage counts are returned as the <em>counts</em> assay in the RSE Full sample metadata is returned as the <em>colData</em> of the RSE</p>
<p>Basic queries include:</p>
<ul>
<li>exon-exon splice junction counts: <em>query_jx</em></li>
<li>exon-level quantifications: <em>query_exon</em></li>
<li>genes-level quantifications: <em>query_gene</em></li>
</ul>
<p>The Gencode v25 annotation defines what genes and exons can be queried (as in <em><a href="https://bioconductor.org/packages/3.14/recount">recount</a></em>).
For splice junctions, both annotated and novel junctions are queried at the same time unless one is explicitly filtered.</p>
<p>Metadata columns will vary by compilation (e.g. TCGA vs. GTEx).
Please be cautious with metadata. We strive to make it as complete and usable as possible, but it can still be incomplete, incorrect or inconsistently formatted (e.g. “age” in the <code>srav2</code> compilation).</p>
<p>Start by querying for all junctions within the region of gene CD99:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="bioconductor.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Query coverage for gene, exon, and annotated junctions across all</span></span>
<span id="cb38-2"><a href="bioconductor.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#in the region of the CD99 gene</span></span>
<span id="cb38-3"><a href="bioconductor.html#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co">#from GTEx v6 sample compilation</span></span>
<span id="cb38-4"><a href="bioconductor.html#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co">#CD99 is chosen for its size</span></span>
<span id="cb38-5"><a href="bioconductor.html#cb38-5" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> <span class="fu">QueryBuilder</span>(<span class="at">compilation=</span><span class="st">&quot;gtex&quot;</span>, <span class="at">regions=</span><span class="st">&quot;CD99&quot;</span>)</span>
<span id="cb38-6"><a href="bioconductor.html#cb38-6" aria-hidden="true" tabindex="-1"></a>cd99.gene <span class="ot">&lt;-</span> <span class="fu">query_gene</span>(sb)</span>
<span id="cb38-7"><a href="bioconductor.html#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cd99.gene)</span></code></pre></div>
<pre><code>## [1]    1 9662</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="bioconductor.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cd99.gene)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 1 9662 
## metadata(1): uri
## assays(1): counts
## rownames(1): 23
## rowData names(12): DataSource:Type snaptron_id ... coverage_median
##   compilation_id
## colnames(9662): rail_50099 rail_50100 ... rail_59759 rail_59760
## colData names(322): rail_id Run ... junction_coverage
##   junction_avg_coverage</code></pre>
<p>Now query for junctions:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="bioconductor.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Query all exon-exon splice junctions within the region of gene CD99</span></span>
<span id="cb42-2"><a href="bioconductor.html#cb42-2" aria-hidden="true" tabindex="-1"></a>cd99.jx.all <span class="ot">&lt;-</span> <span class="fu">query_jx</span>(sb)</span>
<span id="cb42-3"><a href="bioconductor.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cd99.jx.all)</span></code></pre></div>
<pre><code>## [1] 3485 9662</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="bioconductor.html#cb44-1" aria-hidden="true" tabindex="-1"></a>cd99.jx.all</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 3485 9662 
## metadata(1): uri
## assays(1): counts
## rownames(3485): 28340058 28340273 ... 28352407 28352408
## rowData names(12): DataSource:Type snaptron_id ... coverage_median
##   source_dataset_id
## colnames(9662): rail_50099 rail_50100 ... rail_59759 rail_59760
## colData names(322): rail_id Run ... junction_coverage
##   junction_avg_coverage</code></pre>
<p>Now query for junctions and filter by sample type: Brain:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="bioconductor.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co">#now subfilter by sample tissue</span></span>
<span id="cb46-2"><a href="bioconductor.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co">#GTEx samples that are labeled with tissue type &quot;Brain&quot;</span></span>
<span id="cb46-3"><a href="bioconductor.html#cb46-3" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> <span class="fu">set_column_filters</span>(sb, SMTS <span class="sc">==</span> <span class="st">&quot;Brain&quot;</span>)</span>
<span id="cb46-4"><a href="bioconductor.html#cb46-4" aria-hidden="true" tabindex="-1"></a>cd99.jx.all <span class="ot">&lt;-</span> <span class="fu">query_jx</span>(sb)</span>
<span id="cb46-5"><a href="bioconductor.html#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cd99.jx.all)</span></code></pre></div>
<pre><code>## [1]  637 1409</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="bioconductor.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cd99.jx.all)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 6 1409 
## metadata(1): uri
## assays(1): counts
## rownames(6): 28340058 28341126 ... 28341507 28341508
## rowData names(12): DataSource:Type snaptron_id ... coverage_median
##   source_dataset_id
## colnames(1409): rail_50099 rail_50105 ... rail_59748 rail_59753
## colData names(322): rail_id Run ... junction_coverage
##   junction_avg_coverage</code></pre>
<p>Same query/filter again but for exons:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="bioconductor.html#cb50-1" aria-hidden="true" tabindex="-1"></a>cd99.exon <span class="ot">&lt;-</span> <span class="fu">query_exon</span>(sb)</span>
<span id="cb50-2"><a href="bioconductor.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cd99.exon)</span></code></pre></div>
<pre><code>## [1]   32 1409</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="bioconductor.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cd99.exon)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 6 1409 
## metadata(1): uri
## assays(1): counts
## rownames(6): 691 692 ... 695 696
## rowData names(12): DataSource:Type snaptron_id ... coverage_median
##   compilation_id
## colnames(1409): rail_50099 rail_50105 ... rail_59748 rail_59753
## colData names(322): rail_id Run ... junction_coverage
##   junction_avg_coverage</code></pre>
<p>Now query junctions but further filter for only those which are fully annotated:</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="bioconductor.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="do">###Only query junctions which are fully annotated---both left and</span></span>
<span id="cb54-2"><a href="bioconductor.html#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="co">#right splice sites are found together in one or more of the</span></span>
<span id="cb54-3"><a href="bioconductor.html#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Snaptron sourced annotations</span></span>
<span id="cb54-4"><a href="bioconductor.html#cb54-4" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> <span class="fu">set_row_filters</span>(sb, annotated <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb54-5"><a href="bioconductor.html#cb54-5" aria-hidden="true" tabindex="-1"></a>cd99.jx <span class="ot">&lt;-</span> <span class="fu">query_jx</span>(sb)</span>
<span id="cb54-6"><a href="bioconductor.html#cb54-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cd99.jx)</span></code></pre></div>
<pre><code>## [1]   23 1409</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="bioconductor.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cd99.jx)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 6 1409 
## metadata(1): uri
## assays(1): counts
## rownames(6): 28350154 28350172 ... 28350658 28350686
## rowData names(12): DataSource:Type snaptron_id ... coverage_median
##   source_dataset_id
## colnames(1409): rail_50099 rail_50105 ... rail_59748 rail_59753
## colData names(322): rail_id Run ... junction_coverage
##   junction_avg_coverage</code></pre>
<p>Now check an example of the metadata stored in the RSE, InsertSize:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="bioconductor.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Metadata is stored directly in the RSE object.</span></span>
<span id="cb58-2"><a href="bioconductor.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co">#For example the library insert size can be retrieved</span></span>
<span id="cb58-3"><a href="bioconductor.html#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co">#across all runs  in the RSE</span></span>
<span id="cb58-4"><a href="bioconductor.html#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cd99.jx.all<span class="sc">$</span>InsertSize)</span></code></pre></div>
<pre><code>## [1] 162 302 163 150 223 150</code></pre>
</div>
</div>
<div id="megadepth" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> megadepth</h2>
<p><em>The following information is taken from the <a href="https://bioconductor.org/packages/megadepth"><code>megadepth</code></a> documentation.</em></p>
<p>To get started, we need to load the <em><a href="https://bioconductor.org/packages/3.14/megadepth">megadepth</a></em> package into our R session. This will load all the required dependencies.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="bioconductor.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;megadepth&quot;</span>)</span></code></pre></div>
<p>Once we have the R package loaded, we need to install the <a href="https://github.com/ChristopherWilks/megadepth">Megadepth</a> software. We can do so with <code>install_megadepth()</code>, which downloads a binary for your OS (Linux, Windows or macOS).<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> We can then use with an example BigWig file to compute the coverage at a set of regions.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="bioconductor.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Install the latest version of Megadepth</span></span>
<span id="cb61-2"><a href="bioconductor.html#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install_megadepth</span>(<span class="at">force =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## The latest megadepth version is 1.1.1</code></pre>
<pre><code>## This is not an interactive session, therefore megadepth has been installed temporarily to 
## /tmp/RtmptRew4q/megadepth</code></pre>
<p>Next, we might want to use <em><a href="https://bioconductor.org/packages/3.14/megadepth">megadepth</a></em> to quantify the coverage at a set of regions of the genome of interest to us. Here we will use two example files that are include in the package for illustration and testing purposes. One of them is a <a href="https://genome.ucsc.edu/goldenPath/help/bigWig.html">bigWig file</a> that contains the base-pair coverage information for a sample of interest and the second one is <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED file</a> which contains the genomic region coordinates of interest. So we first locate them.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="bioconductor.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Next, we locate the example BigWig and annotation files</span></span>
<span id="cb64-2"><a href="bioconductor.html#cb64-2" aria-hidden="true" tabindex="-1"></a>example_bw <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;tests&quot;</span>, <span class="st">&quot;test.bam.all.bw&quot;</span>,</span>
<span id="cb64-3"><a href="bioconductor.html#cb64-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;megadepth&quot;</span>, <span class="at">mustWork =</span> <span class="cn">TRUE</span></span>
<span id="cb64-4"><a href="bioconductor.html#cb64-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb64-5"><a href="bioconductor.html#cb64-5" aria-hidden="true" tabindex="-1"></a>annotation_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;tests&quot;</span>, <span class="st">&quot;testbw2.bed&quot;</span>,</span>
<span id="cb64-6"><a href="bioconductor.html#cb64-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;megadepth&quot;</span>, <span class="at">mustWork =</span> <span class="cn">TRUE</span></span>
<span id="cb64-7"><a href="bioconductor.html#cb64-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb64-8"><a href="bioconductor.html#cb64-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-9"><a href="bioconductor.html#cb64-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Where are they?</span></span>
<span id="cb64-10"><a href="bioconductor.html#cb64-10" aria-hidden="true" tabindex="-1"></a>example_bw</span></code></pre></div>
<pre><code>## [1] &quot;/usr/local/lib/R/host-site-library/megadepth/tests/test.bam.all.bw&quot;</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="bioconductor.html#cb66-1" aria-hidden="true" tabindex="-1"></a>annotation_file</span></code></pre></div>
<pre><code>## [1] &quot;/usr/local/lib/R/host-site-library/megadepth/tests/testbw2.bed&quot;</code></pre>
<p>Once we have located the example files we can proceed to calculating the base-pair coverage for our genomic regions of interest. There are several ways to do this with <em><a href="https://bioconductor.org/packages/3.14/megadepth">megadepth</a></em>, but here we use the user-friendly function <code>get_coverage()</code>. This function will perform a given operation <strong>op</strong> on the bigWig file for a given set of regions of interest (<em>annotation</em>). One of those operations is to compute the mean base-pair coverage for each input region. This is what we’ll do with our example bigWig file.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="bioconductor.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We can then use megadepth to compute the coverage</span></span>
<span id="cb68-2"><a href="bioconductor.html#cb68-2" aria-hidden="true" tabindex="-1"></a>bw_cov <span class="ot">&lt;-</span> <span class="fu">get_coverage</span>(</span>
<span id="cb68-3"><a href="bioconductor.html#cb68-3" aria-hidden="true" tabindex="-1"></a>    example_bw,</span>
<span id="cb68-4"><a href="bioconductor.html#cb68-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">op =</span> <span class="st">&quot;mean&quot;</span>,</span>
<span id="cb68-5"><a href="bioconductor.html#cb68-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation =</span> annotation_file</span>
<span id="cb68-6"><a href="bioconductor.html#cb68-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb68-7"><a href="bioconductor.html#cb68-7" aria-hidden="true" tabindex="-1"></a>bw_cov</span></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 1 metadata column:
##         seqnames          ranges strand |     score
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]      chr10            0-10      * |      0.00
##   [2]      chr10 8756697-8756762      * |     15.85
##   [3]      chr10 4359156-4359188      * |      3.00
##   [4] GL000219.1   168500-168620      * |      1.26
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p><code>get_coverage()</code> returns an object that is familiar to <em><a href="https://bioconductor.org/packages/3.14/GenomicRanges">GenomicRanges</a></em> users, that is, a <code>GRanges</code> object that can be used with other Bioconductor software packages.</p>
<p>This example is just the tip of the iceberg, as <a href="https://github.com/ChristopherWilks/megadepth">Megadepth</a> and thus <em><a href="https://bioconductor.org/packages/3.14/megadepth">megadepth</a></em> can do a lot of useful processing operations on BAM and bigWig files.</p>
<p>Please continue to the <a href="https://bioconductor.org/packages/megadepth">users guide</a> for more detailed information.</p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="2">
<li id="fn2"><p>Please check <a href="https://github.com/ChristopherWilks/megadepth">Megadepth</a> for instructions on how to compile the software from source if the binary version doesn’t work for you.<a href="bioconductor.html#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quality-check-fields.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="raw-files.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
