<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Quality check fields | recount3: human and mouse public RNA-seq data</title>
  <meta name="description" content="4 Quality check fields | recount3: human and mouse public RNA-seq data" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Quality check fields | recount3: human and mouse public RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Quality check fields | recount3: human and mouse public RNA-seq data" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="how-to-ask-for-help.html"/>
<link rel="next" href="bioconductor.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="css/styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="part"><span><b>I Get RNA-seq data!</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#human"><i class="fa fa-check"></i><b>1.1</b> Human</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#mouse"><i class="fa fa-check"></i><b>1.2</b> Mouse</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#study-explorer"><i class="fa fa-check"></i><b>1.3</b> Study explorer</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#how-to-cite-recount3"><i class="fa fa-check"></i><b>1.4</b> How to cite <code>recount3</code></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quick-access.html"><a href="quick-access.html"><i class="fa fa-check"></i><b>2</b> Quick Access</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-access.html"><a href="quick-access.html#quick-recount3"><i class="fa fa-check"></i><b>2.1</b> Quick recount3</a></li>
<li class="chapter" data-level="2.2" data-path="quick-access.html"><a href="quick-access.html#quick-snapcount"><i class="fa fa-check"></i><b>2.2</b> Quick snapcount</a></li>
<li class="chapter" data-level="2.3" data-path="quick-access.html"><a href="quick-access.html#quick-raw-files"><i class="fa fa-check"></i><b>2.3</b> Quick raw files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html"><i class="fa fa-check"></i><b>3</b> How to ask for help</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#gitter"><i class="fa fa-check"></i><b>3.1</b> Gitter</a></li>
<li class="chapter" data-level="3.2" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#bioconductor-support-website"><i class="fa fa-check"></i><b>3.2</b> Bioconductor Support website</a></li>
<li class="chapter" data-level="3.3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#github-issues"><i class="fa fa-check"></i><b>3.3</b> GitHub Issues</a></li>
</ul></li>
<li class="part"><span><b>II Details</b></span></li>
<li class="chapter" data-level="4" data-path="quality-check-fields.html"><a href="quality-check-fields.html"><i class="fa fa-check"></i><b>4</b> Quality check fields</a></li>
<li class="chapter" data-level="5" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>5</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="5.1" data-path="bioconductor.html"><a href="bioconductor.html#recount3"><i class="fa fa-check"></i><b>5.1</b> recount3</a></li>
<li class="chapter" data-level="5.2" data-path="bioconductor.html"><a href="bioconductor.html#snapcount"><i class="fa fa-check"></i><b>5.2</b> snapcount</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="bioconductor.html"><a href="bioconductor.html#basic-queries"><i class="fa fa-check"></i><b>5.2.1</b> Basic queries</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="bioconductor.html"><a href="bioconductor.html#megadepth"><i class="fa fa-check"></i><b>5.3</b> megadepth</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="raw-files.html"><a href="raw-files.html"><i class="fa fa-check"></i><b>6</b> Raw Files</a>
<ul>
<li class="chapter" data-level="6.1" data-path="raw-files.html"><a href="raw-files.html#data-source-vs-collection"><i class="fa fa-check"></i><b>6.1</b> Data source vs collection</a></li>
<li class="chapter" data-level="6.2" data-path="raw-files.html"><a href="raw-files.html#annotation-files"><i class="fa fa-check"></i><b>6.2</b> Annotation files</a></li>
<li class="chapter" data-level="6.3" data-path="raw-files.html"><a href="raw-files.html#project-level-count-files"><i class="fa fa-check"></i><b>6.3</b> Project-level count files</a></li>
<li class="chapter" data-level="6.4" data-path="raw-files.html"><a href="raw-files.html#project-level-metadata-files"><i class="fa fa-check"></i><b>6.4</b> Project-level metadata files</a></li>
<li class="chapter" data-level="6.5" data-path="raw-files.html"><a href="raw-files.html#sample-level-bigwig-files"><i class="fa fa-check"></i><b>6.5</b> Sample-level BigWig files</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="related-software.html"><a href="related-software.html"><i class="fa fa-check"></i><b>7</b> Related software</a>
<ul>
<li class="chapter" data-level="7.1" data-path="related-software.html"><a href="related-software.html#beyond-recount3"><i class="fa fa-check"></i><b>7.1</b> Beyond recount3</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="related-analyses.html"><a href="related-analyses.html"><i class="fa fa-check"></i><b>8</b> Related analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="related-analyses.html"><a href="related-analyses.html#beyond-recount3-1"><i class="fa fa-check"></i><b>8.1</b> Beyond recount3</a></li>
</ul></li>
<li class="part"><span><b>III Expand <code>ReCount</code></b></span></li>
<li class="chapter" data-level="9" data-path="your-data.html"><a href="your-data.html"><i class="fa fa-check"></i><b>9</b> Your data</a>
<ul>
<li class="chapter" data-level="9.1" data-path="your-data.html"><a href="your-data.html#process-your-data"><i class="fa fa-check"></i><b>9.1</b> Process your data</a></li>
<li class="chapter" data-level="9.2" data-path="your-data.html"><a href="your-data.html#contribute-your-collections"><i class="fa fa-check"></i><b>9.2</b> Contribute your collections</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="your-data.html"><a href="your-data.html#collection-metadata"><i class="fa fa-check"></i><b>9.2.1</b> Collection metadata</a></li>
<li class="chapter" data-level="9.2.2" data-path="your-data.html"><a href="your-data.html#collection-project-location"><i class="fa fa-check"></i><b>9.2.2</b> Collection project location</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="join-the-recount-team.html"><a href="join-the-recount-team.html"><i class="fa fa-check"></i><b>10</b> Join the <code>ReCount</code> team!</a></li>
<li class="chapter" data-level="" data-path="r-session-information.html"><a href="r-session-information.html"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">recount3: human and mouse public RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-check-fields" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Quality check fields</h1>
<p>We used a number of tools to collect potentially useful quality-control (QC) measures. Specifically, we used <code>seqtk</code> <span class="citation">(<a href="references.html#ref-seqtk" role="doc-biblioref">Li 2020 (accessed August 18, 2020)</a>)</span>, the <code>idxstats</code> subcommand of <code>samtools</code>, the output of <code>STAR</code>, our own <code>megadepth</code> tool, and <code>featureCounts</code>. We examine each in turn, listing the specific QC measures calculated be each.</p>
<p>Monorail runs the <code>seqtk fqchk</code> command on input FASTQ files to collect base-quality and base-composition summaries for all sequencing cycles. We distill these into a few QC measures included with every summarized run in <code>recount3</code>.</p>
<ol style="list-style-type: decimal">
<li><strong><code>min_len</code></strong>: minimum read length</li>
<li><strong><code>max_len</code></strong>: maximum read length</li>
<li><strong><code>avg_len</code></strong>: average read length</li>
<li><strong><code>#distinct_quality_values</code></strong>: number of different quality scores present in the sequence run’s base qualities</li>
<li><strong><code>#bases</code></strong>: total number of bases across spots (not including both read mates if paired)</li>
<li><strong><code>%A</code></strong>: percent of bases that are A</li>
<li><strong><code>%C</code></strong>: percent of bases that are C</li>
<li><strong><code>%G</code></strong>: percent of bases that are G</li>
<li><strong><code>%T</code></strong>: percent of bases that are T</li>
<li><strong><code>%N</code></strong>: percent of bases that are N</li>
<li><strong><code>avgQ</code></strong>: weighted average over Phred quality scores present in the sequence run, where weights are the Phred quality values themselves</li>
<li><strong><code>errQ</code></strong>: negatively scaled log of the weighted average Phred quality scores present in the sequence run, where weights are the error probabilities associated with the Phred quality scores</li>
</ol>
<p>Monorail uses <code>STAR</code> to align RNA-seq reads in a spliced fashion to a reference genome, without using any annotation. Files output by <code>STAR</code>, particularly the <code>Log.out</code> and <code>Log.final.out</code>, report a number of measures that can be used for QC. We compile these into a number of QC measures included with every summarized run in recount3. Note that some of these measures are reported separately for the two ends of a paired-end read. We omit the second-end versions of these QC measures here for space reasons.</p>
<p>From the <code>STAR</code> manual (version 2.7.2b):</p>
<blockquote>
<blockquote>
<p>Log.final.out: summary mapping statistics after mapping job is complete, very useful for quality control. The statistics are calculated for each read (single- or paired-end) and then summed or averaged over all reads. Note that STAR counts a paired-end read as one read, (unlike the samtools flagstat/idxstats, which count each mate separately). Most of the information is collected about the UNIQUE mappers (unlike samtools flagstat/idxstats which does not separate unique or multi-mappers). Each splicing is counted in the numbers of splices, which would correspond to summing the counts in SJ.out.tab. The mismatch/indel error rates are calculated on a per base basis, i.e. as total number of mismatches/indels in all unique mappers divided by the total number of mapped bases.</p>
</blockquote>
</blockquote>
<p>Some of the following definitions include text from the <code>STAR</code> manual/source code, reprinted here for convenience. Please see the <a href="https://github.com/alexdobin/STAR/tree/master/doc"><code>STAR</code> manual</a> for more in depth information.</p>
<ol style="list-style-type: decimal">
<li><strong><code>%_of_chimeric_reads</code></strong>: Number of chimeric reads divided by number of input reads</li>
<li><strong><code>%_of_reads_mapped_to_multiple_loci</code></strong>: Number of reads mapped to multiple loci divided by number of input reads</li>
<li><strong><code>%_of_reads_mapped_to_too_many_loci</code></strong>: Number of reads mapped to (&gt; 10) loci divided by number of input reads</li>
<li><strong><code>%_of_reads_unmapped:_other</code></strong>: Reads are unmapped due to no acceptable seed/windows divided by number of input reads</li>
<li><strong><code>%_of_reads_unmapped:_too_many_mismatches</code></strong>: Number of reads where best alignment has more mismatches than max allowed number of mismatches divided by number of input reads</li>
<li><strong><code>%_of_reads_unmapped:_too_short</code></strong>: Number of reads where best alignment was shorter than min allowed mapped length divided by number of input reads</li>
<li><strong><code>all_mapped_reads</code></strong>: Total number of reads aligned</li>
<li><strong><code>average_input_read_length</code></strong>: Average length of a read</li>
<li><strong><code>average_mapped_length</code></strong>: Average length of an alignment</li>
<li><strong><code>deletion_average_length</code></strong>: Average length of a genomic deletion, i.e. genomic gaps</li>
<li><strong><code>deletion_rate_per_base</code></strong>: Genomic deletions per mapped base</li>
<li><strong><code>insertion_average_length</code></strong>: Average length of a genomic insertion, i.e. read gaps</li>
<li><strong><code>insertion_rate_per_base</code></strong>: Genomics insertions per mapped base</li>
<li><strong><code>mapping_speed_million_of_reads_per_hour</code></strong>: How fast it was to align this sample</li>
<li><strong><code>mismatch_rate_per_base_%</code></strong>: Mismatches per mapped base</li>
<li><strong><code>number_of_chimeric_reads</code></strong>: Total number of reads which were fragmented on aligning, e.g. fusion potential reads</li>
<li><strong><code>number_of_input_reads</code></strong>: Total number of reads input to <code>STAR</code></li>
<li><strong><code>number_of_reads_mapped_to_multiple_loci</code></strong>: Number of reads mapped to multiple loci</li>
<li><strong><code>number_of_reads_mapped_to_too_many_loci</code></strong>: Number of reads mapped to (&gt; 10) loci</li>
<li><strong><code>number_of_reads_unmapped:_other</code></strong>: Number of reads left unmapped due to no acceptable seed/windows</li>
<li><strong><code>number_of_reads_unmapped:_too_many_mismatches</code></strong>: Number of reads where best alignment has more mismatches than max allowed number of mismatches</li>
<li><strong><code>number_of_reads_unmapped:_too_short</code></strong>: Number of reads where best alignment was shorter than min allowed mapped length</li>
<li><strong><code>number_of_splices:_at/ac</code></strong>: Number of canonical splices of AT-AC (and reverse)</li>
<li><strong><code>number_of_splices:_annotated_(sjdb)</code></strong>: Number of splices found that were also in the annotation database</li>
<li><strong><code>number_of_splices:_gc/ag</code></strong>: Number of canonical splices of GC-AG (and reverse)</li>
<li><strong><code>number_of_splices:_gt/ag</code></strong>: Number of canonical splices of GT-AG (and reverse)</li>
<li><strong><code>number_of_splices:_non-canonical</code></strong>: Number of non-canonical splices, anything not GT-AG, AT-AC, GC-AG, or their reverse complement</li>
<li><strong><code>number_of_splices:_total</code></strong>: Total number of splices</li>
<li><strong><code>uniquely_mapped_reads_%</code></strong>: Number of reads which mapped to a single locus divided by number of input reads</li>
<li><strong><code>uniquely_mapped_reads_number</code></strong>: Number of reads which mapped to a single locus</li>
</ol>
<p>Monorail runs the <code>samtools idxstats</code> on the BAM file output by <code>STAR</code> to collect statistics about how many reads aligned to each chromosome in the genome assembly. This can be helpful in, for instance, confirming the sex of the individual sequenced based on alignments to sex chromosomes, or measuring effectiveness of ribosomal RNA depletion by considering the fraction of reads aligned to the mitochondrial genome. We compile these into a number of QC measures included with every summarized run in recount3:</p>
<ol style="list-style-type: decimal">
<li><strong><code>aligned_reads%.chrm</code></strong>: Percent of reads aligning to the mitochondrial genome.</li>
<li><strong><code>aligned_reads%.chrx</code></strong>: Percent of reads aligning to chromosome X.</li>
<li><strong><code>aligned_reads%.chry</code></strong>: Precent of reads aligning to chromosome Y.</li>
</ol>
<p>Monorail runs our <code>megadepth</code> tool on the BAM files output by <code>STAR</code>. The chief function is to convert BAM files to bigWig files that are then added to the recount3 archive. As <code>megadepth</code> performs this conversion, it also summarizes the amount of sequencing coverage within the intervals of a provided BED file representing a gene annotation. These quantifications can be useful for quality control, tell us, for example, what fraction of the coverage is within annotated genes.</p>
<p>Fragment length distribution is based on a special read filter only applied for this purpose to be compatible with CSAW’s fragment counting approach <span class="citation">(<a href="references.html#ref-csaw" role="doc-biblioref">Lun and Smyth 2016</a>)</span>, paired reads in a passing fragment must not be secondary, supplementary, have conflicting read order, be unmapped or be mapped on more than one chromosome. The <code>bc_</code> prefix here refers to the previous name of the Megadepth tool.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<ol style="list-style-type: decimal">
<li><strong><code>bc_auc.all_reads_all_bases</code></strong>: Area under coverage (total depth of coverage evaluated at all bases) for all alignments</li>
<li><strong><code>bc_auc.all_reads_annotated_bases</code></strong>: Area under coverage for all alignments, but only for bases in annotated exons</li>
<li><strong><code>bc_auc.unique_reads_all_bases</code></strong>: Area under coverage for uniquely aligned reads</li>
<li><strong><code>bc_auc.unique_reads_annotated_bases</code></strong>: Area under coverage for uniquely aligned reads, but only for bases in annotated exons</li>
<li><strong><code>bc_auc.all_%</code></strong>: <code>bc_auc.all_reads_annotated_bases</code> divided by <code>bc_auc.all_reads_all_bases</code></li>
<li><strong><code>bc_auc.unique_%</code></strong>: <code>bc_auc.unique_reads_annotated_bases</code> divided by <code>bc_auc.unique_reads_all_bases</code></li>
<li><strong><code>bc_frag.count</code></strong>: Total number of read fragments in BAM after filtering</li>
<li><strong><code>bc_frag.kallisto_count</code></strong>: Number of read fragments (&lt; 1000) bp in length in BAM after filtering</li>
<li><strong><code>bc_frag.kallisto_mean_length</code></strong>: Mean length of read fragments (&lt; 1000) bp in length in BAM after filtering</li>
<li><strong><code>bc_frag.mean_length</code></strong>: Mean length of all read fragments in BAM after filtering</li>
<li><strong><code>bc_frag.mode_length</code></strong>: Mode of the read fragment length of all fragments in BAM after filtering</li>
<li><strong><code>bc_frag.mode_length_count</code></strong>: Number of read fraqments with the <code>bc_frag.mode_length</code> in BAM after filtering</li>
</ol>
<p>Finally, Monorail runs <code>featureCounts</code> on the BAM files output by <code>STAR</code>. This provides a <code>second opinion</code> on the quantifications produced by <code>megadepth</code>. While we have not yet found compelling examples where the <code>megadepth</code> and <code>featureCounts</code> outputs disagree, we keep summaries of the <code>featureCounts</code> quantifications as potential QC measures.</p>
<ol style="list-style-type: decimal">
<li><strong><code>exon_fc.all_%</code></strong>: <code>exon_fc_count_all.assigned</code> divided by <code>all_mapped_reads</code> (from <code>STAR</code>)</li>
<li><strong><code>exon_fc.unique_%</code></strong>: <code>exon_fc_count_unique.assigned</code> divided by <code>uniquely_mapped_reads_number</code> (from <code>STAR</code>)</li>
<li><strong><code>exon_fc_count_all.total</code></strong>: Total number of fragments, including multi-mappers, input to <code>featureCounts</code></li>
<li><strong><code>exon_fc_count_all.assigned</code></strong>: Number of fragments, including multi-mappers, assigned by <code>featureCounts</code> to an exon</li>
<li><strong><code>exon_fc_count_unique.total</code></strong>: Total number of uniquely mapping fragments input to <code>featureCounts</code></li>
<li><strong><code>exon_fc_count_unique.assigned</code></strong>: Number of uniquely mapping fragments assigned by <code>featureCounts</code> to an exon</li>
<li><strong><code>gene_fc.all_%</code></strong>: <code>gene_fc_count_all.assigned</code> divided by <code>all_mapped_read</code>s (from <code>STAR</code>)</li>
<li><strong><code>gene_fc.unique_%</code></strong>: <code>gene_fc_count_unique.assigned</code> divided by <code>uniquely_mapped_reads_number</code> (from <code>STAR</code>)</li>
<li><strong><code>gene_fc_count_all.total</code></strong>: Total number of fragments, including multi-mappers, input to <code>featureCounts</code></li>
<li><strong><code>gene_fc_count_all.assigned</code></strong>: Number of fragments, including multi-mappers, assigned by <code>featureCounts</code> to a gene</li>
<li><strong><code>gene_fc_count_unique.total</code></strong>: Total number of uniquely mapping fragments input to <code>featureCounts</code></li>
<li><strong><code>gene_fc_count_unique.assigned</code></strong>: Number of uniquely mapping fragments assigned by <code>featureCounts</code> to a gene</li>
</ol>

</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>Megadepth used to be called BamCount.<a href="quality-check-fields.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-to-ask-for-help.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bioconductor.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
