<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Your data | recount3: human and mouse public RNA-seq data</title>
  <meta name="description" content="9 Your data | recount3: human and mouse public RNA-seq data" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Your data | recount3: human and mouse public RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Your data | recount3: human and mouse public RNA-seq data" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="related-analyses.html"/>
<link rel="next" href="join-the-recount-team.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="css/styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="part"><span><b>I Get RNA-seq data!</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#human"><i class="fa fa-check"></i><b>1.1</b> Human</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#mouse"><i class="fa fa-check"></i><b>1.2</b> Mouse</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#study-explorer"><i class="fa fa-check"></i><b>1.3</b> Study explorer</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#how-to-cite-recount3"><i class="fa fa-check"></i><b>1.4</b> How to cite <code>recount3</code></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quick-access.html"><a href="quick-access.html"><i class="fa fa-check"></i><b>2</b> Quick Access</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-access.html"><a href="quick-access.html#quick-recount3"><i class="fa fa-check"></i><b>2.1</b> Quick recount3</a></li>
<li class="chapter" data-level="2.2" data-path="quick-access.html"><a href="quick-access.html#quick-snapcount"><i class="fa fa-check"></i><b>2.2</b> Quick snapcount</a></li>
<li class="chapter" data-level="2.3" data-path="quick-access.html"><a href="quick-access.html#quick-raw-files"><i class="fa fa-check"></i><b>2.3</b> Quick raw files</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html"><i class="fa fa-check"></i><b>3</b> How to ask for help</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#gitter"><i class="fa fa-check"></i><b>3.1</b> Gitter</a></li>
<li class="chapter" data-level="3.2" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#bioconductor-support-website"><i class="fa fa-check"></i><b>3.2</b> Bioconductor Support website</a></li>
<li class="chapter" data-level="3.3" data-path="how-to-ask-for-help.html"><a href="how-to-ask-for-help.html#github-issues"><i class="fa fa-check"></i><b>3.3</b> GitHub Issues</a></li>
</ul></li>
<li class="part"><span><b>II Details</b></span></li>
<li class="chapter" data-level="4" data-path="quality-check-fields.html"><a href="quality-check-fields.html"><i class="fa fa-check"></i><b>4</b> Quality check fields</a></li>
<li class="chapter" data-level="5" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>5</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="5.1" data-path="bioconductor.html"><a href="bioconductor.html#recount3"><i class="fa fa-check"></i><b>5.1</b> recount3</a></li>
<li class="chapter" data-level="5.2" data-path="bioconductor.html"><a href="bioconductor.html#snapcount"><i class="fa fa-check"></i><b>5.2</b> snapcount</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="bioconductor.html"><a href="bioconductor.html#basic-queries"><i class="fa fa-check"></i><b>5.2.1</b> Basic queries</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="bioconductor.html"><a href="bioconductor.html#megadepth"><i class="fa fa-check"></i><b>5.3</b> megadepth</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="raw-files.html"><a href="raw-files.html"><i class="fa fa-check"></i><b>6</b> Raw Files</a>
<ul>
<li class="chapter" data-level="6.1" data-path="raw-files.html"><a href="raw-files.html#data-source-vs-collection"><i class="fa fa-check"></i><b>6.1</b> Data source vs collection</a></li>
<li class="chapter" data-level="6.2" data-path="raw-files.html"><a href="raw-files.html#annotation-files"><i class="fa fa-check"></i><b>6.2</b> Annotation files</a></li>
<li class="chapter" data-level="6.3" data-path="raw-files.html"><a href="raw-files.html#project-level-count-files"><i class="fa fa-check"></i><b>6.3</b> Project-level count files</a></li>
<li class="chapter" data-level="6.4" data-path="raw-files.html"><a href="raw-files.html#project-level-metadata-files"><i class="fa fa-check"></i><b>6.4</b> Project-level metadata files</a></li>
<li class="chapter" data-level="6.5" data-path="raw-files.html"><a href="raw-files.html#sample-level-bigwig-files"><i class="fa fa-check"></i><b>6.5</b> Sample-level BigWig files</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="related-software.html"><a href="related-software.html"><i class="fa fa-check"></i><b>7</b> Related software</a>
<ul>
<li class="chapter" data-level="7.1" data-path="related-software.html"><a href="related-software.html#beyond-recount3"><i class="fa fa-check"></i><b>7.1</b> Beyond recount3</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="related-analyses.html"><a href="related-analyses.html"><i class="fa fa-check"></i><b>8</b> Related analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="related-analyses.html"><a href="related-analyses.html#beyond-recount3-1"><i class="fa fa-check"></i><b>8.1</b> Beyond recount3</a></li>
</ul></li>
<li class="part"><span><b>III Expand <code>ReCount</code></b></span></li>
<li class="chapter" data-level="9" data-path="your-data.html"><a href="your-data.html"><i class="fa fa-check"></i><b>9</b> Your data</a>
<ul>
<li class="chapter" data-level="9.1" data-path="your-data.html"><a href="your-data.html#process-your-data"><i class="fa fa-check"></i><b>9.1</b> Process your data</a></li>
<li class="chapter" data-level="9.2" data-path="your-data.html"><a href="your-data.html#contribute-your-collections"><i class="fa fa-check"></i><b>9.2</b> Contribute your collections</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="your-data.html"><a href="your-data.html#collection-metadata"><i class="fa fa-check"></i><b>9.2.1</b> Collection metadata</a></li>
<li class="chapter" data-level="9.2.2" data-path="your-data.html"><a href="your-data.html#collection-project-location"><i class="fa fa-check"></i><b>9.2.2</b> Collection project location</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="join-the-recount-team.html"><a href="join-the-recount-team.html"><i class="fa fa-check"></i><b>10</b> Join the <code>ReCount</code> team!</a></li>
<li class="chapter" data-level="" data-path="r-session-information.html"><a href="r-session-information.html"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">recount3: human and mouse public RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="your-data" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Your data</h1>
<p><em>Note that this section will need to be refined once we have gone through this process. Currently we are focused on releasing the initial version of recount3.</em></p>
<div id="process-your-data" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Process your data</h2>
<p>While <a href="https://github.com/langmead-lab/recount-pump"><code>recount_pump</code></a> contains code for generating recount3, if you want to use the Monorail RNA-seq processing pipeline (alignment/quantification) for your own data, we highly recommend that you start by looking at the <a href="https://github.com/langmead-lab/monorail-external/"><strong><code>monorail-external</code></strong></a> GitHub repository. It contains a very detailed README with instructions on how to install the software, download the annotation files, and process raw RNA-seq data (with some example files used for illustration purposes).</p>
<p>In order to add your data to recount3, we will need all the <a href="raw-files.html#raw-files">raw files</a> for your study. That is:</p>
<ul>
<li>the gene count text files for each annotation we support (organism-dependent)</li>
<li>similarly, the exon count text files for each annotation</li>
<li>the exon-exon junction files<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a></li>
<li>the five metadata files</li>
<li>the bigWig files for each sample</li>
</ul>
<p>We will then need to organize your data into the directory structure expected by <code>recount3</code>.</p>
</div>
<div id="contribute-your-collections" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Contribute your collections</h2>
<p>As described in the <a href="raw-files.html#raw-files">raw files</a> section, collections involve creating a metadata file for a custom set of samples across one or more studies (typically more than one study). You can find some example collections at <a href="http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/" class="uri">http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/</a>. For example, <a href="http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/" class="uri">http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/</a> contains two files:</p>
<ul>
<li><code>&lt;collection_name&gt;</code>.custom.gz: gtex_geuvadis.custom.gz</li>
<li><code>&lt;collection_name&gt;</code>.recount_project.gz: gtex_geuvadis.recount_project.gz (optional!)</li>
</ul>
<p>These files include the 3 columns we use internally for identifying all samples:</p>
<ul>
<li><code>rail_id</code>: used by Snaptron</li>
<li><code>study</code>: the project name. See <code>recount3::available_projects()</code> for supported options (or use your own custom <code>recount3_url</code>).</li>
<li><code>external_id</code>: typically the SRA run ID but each data source has different unique IDs, like TCGA which uses much longer sample IDs.</li>
</ul>
<div id="collection-metadata" class="section level3" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Collection metadata</h3>
<p>The <code>&lt;collection_name&gt;</code>.custom.gz can then contain any additional columns of interest. For example, in the <a href="https://github.com/LieberInstitute/recount-brain"><code>recount-brain</code> project</a> we manually curated samples from 62 studies and standardized variables across the 62 studies.</p>
<p>Here’s how the example collection metadata file looks like:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="your-data.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">read.delim</span>(</span>
<span id="cb75-2"><a href="your-data.html#cb75-2" aria-hidden="true" tabindex="-1"></a>    recount3<span class="sc">::</span><span class="fu">file_retrieve</span>(</span>
<span id="cb75-3"><a href="your-data.html#cb75-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/gtex_geuvadis.custom.gz&quot;</span></span>
<span id="cb75-4"><a href="your-data.html#cb75-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb75-5"><a href="your-data.html#cb75-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## 2021-09-13 21:20:48 caching file gtex_geuvadis.custom.gz.</code></pre>
<pre><code>## adding rname &#39;http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/gtex_geuvadis.custom.gz&#39;</code></pre>
<pre><code>##   rail_id               external_id     study sequencing_type  tissue
## 1   98371                 ERR188021 ERP001942            bulk     LCL
## 2  153881 GTEX-U3ZM-0826-SM-4DXU6.1   BLADDER            bulk BLADDER</code></pre>
<p>This file has to be readble with R using the following code:</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="your-data.html#cb79-1" aria-hidden="true" tabindex="-1"></a>utils<span class="sc">::</span><span class="fu">read.delim</span>(</span>
<span id="cb79-2"><a href="your-data.html#cb79-2" aria-hidden="true" tabindex="-1"></a>    file_path,</span>
<span id="cb79-3"><a href="your-data.html#cb79-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb79-4"><a href="your-data.html#cb79-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb79-5"><a href="your-data.html#cb79-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">quote =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb79-6"><a href="your-data.html#cb79-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">comment.char =</span> <span class="st">&quot;&quot;</span></span>
<span id="cb79-7"><a href="your-data.html#cb79-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>Since this is the main file that needs to be produced for adding a collection to <code>recount3</code>, if your collection involves data that is already present in <code>recount3</code>, it will be very easy for us to add it to our resource. Otherwise we will need the <a href="raw-files.html#raw-files">raw files</a> for the corresponding new data.</p>
</div>
<div id="collection-project-location" class="section level3" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Collection project location</h3>
<p>The second text file specifies the required information for locating the samples. While we had considered using this type of file, we don’t require it anymore. For historical purposes, this is how that file looked:<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="your-data.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">read.delim</span>(</span>
<span id="cb80-2"><a href="your-data.html#cb80-2" aria-hidden="true" tabindex="-1"></a>    recount3<span class="sc">::</span><span class="fu">file_retrieve</span>(</span>
<span id="cb80-3"><a href="your-data.html#cb80-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/gtex_geuvadis.recount_project.gz&quot;</span></span>
<span id="cb80-4"><a href="your-data.html#cb80-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb80-5"><a href="your-data.html#cb80-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## 2021-09-13 21:20:48 caching file gtex_geuvadis.recount_project.gz.</code></pre>
<pre><code>## adding rname &#39;http://snaptron.cs.jhu.edu/data/temp/recount3/human/collections/gtex_geuvadis/metadata/gtex_geuvadis.recount_project.gz&#39;</code></pre>
<pre><code>##   rail_id               external_id     study   project     organism
## 1   98371                 ERR188021 ERP001942 ERP001942 Homo sapiens
## 2  153881 GTEX-U3ZM-0826-SM-4DXU6.1   BLADDER   BLADDER Homo sapiens
##         file_source           metadata_source date_processed
## 1  data_sources/sra collections/gtex_geuvadis     2019-10-01
## 2 data_sources/gtex collections/gtex_geuvadis     2019-11-01</code></pre>

</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="5">
<li id="fn5"><p>Hm… I think this will be tricky if new exon-exon junctions are found in someone’s data which are absent in recount3. This is VERY likely to happen!<a href="your-data.html#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>Once we have more collections I can verify this!<a href="your-data.html#fnref6" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="related-analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="join-the-recount-team.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
